### ISSUE TITLE
OpKernel   for unknown op: HashTable  And   incompatible with expected float_ref

### ISSUE THREAD

**Issue by jw2100** [tensorflow/tensorflow#17983]
OS Platform : Cestos7
python:2.7
gcc:4.8
TensorFlow installed from source 
tf version :  ('v1.4.0-19-ga52c8d9', '1.4.1')

-----------------------------------------------

2018-03-25 07:49:05.497590: E tensorflow/core/framework/op_kernel.cc:1197] OpKernel ('op: "LeftShift" device_type: "CPU" constraint { name: "T" allowed_values { list { type: DT_INT8 } } }') for unknown op: LeftShift
2018-03-25 07:49:05.497623: E tensorflow/core/framework/op_kernel.cc:1197] OpKernel ('op: "LookupTableInsert" device_type: "CPU"') for unknown op: LookupTableInsert
2018-03-25 07:49:05.497679: E tensorflow/core/framework/op_kernel.cc:1197] OpKernel ('op: "HashTable" device_type: "CPU" constraint { name: "key_dtype" allowed_values { list { type: DT_INT32 } } } constraint { name: "value_dtype" allowed_values { list { type: DT_INT32 } } }') for unknown op: HashTable

---------------------------------------------

Invalid argument: Input 0 of node loss/transitions/Assign was passed float from loss/transitions:0 incompatible with expected float_ref.

---------------------------------------------

**In my model**: 

all variable is similar

`with tf.variable_scope("softmax") as scope:
            self.W = tf.get_variable(
                shape=[hidden_dim * 2, num_classes], 
                initializer=tf.truncated_normal_initializer(stddev=0.01), 
                name="weights",
                regularizer=tf.contrib.layers.l2_regularizer(self.l2_reg_lambda))`

**In freeze**, 
`
 45     with tf.Session(graph=tf.Graph()) as sess:

 47         saver = tf.train.import_meta_graph(input_checkpoint + '.meta', clear_devices=clear_devices)

 48         saver.restore(sess, input_checkpoint)

 50         graph = tf.get_default_graph().as_graph_def()
 52         if graph_pb:
 53             new_graph = graph_pb2.GraphDef()
 54             with tf.gfile.FastGFile(graph_pb, 'rb') as f:
 55                 new_graph.ParseFromString(f.read())
 56             graph = new_graph

 69         output_graph_def = tf.graph_util.convert_variables_to_constants(
 70             sess,
 71             graph,
 72             output_node_names.split(",") 
 73         )

 76         tf.train.write_graph(output_graph_def, output_dir, 'freeze_graph.pb', as_text=False)

 77         tf.train.write_graph(output_graph_def, output_dir, 'freeze_graph.txt', as_text=True)
`

and _**in c++**_ is :
`
Session* session;

Status status = NewSession(SessionOptions(), &session);

GraphDef graph_def;

status = ReadBinaryProto(Env::Default(), "../demo/deep_model/freeze_graph.pb", &graph_def);

status = session->Create(graph_def);
`

----------------------------------------------------

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by jw2100** [comment]
@tensorflowbutler     have updated
